src/Adapter/MSTest.TestAdapter/Discovery/UnitTestDiscoverer.cs

```diff
@@ -109,13 +112,25 @@ internal void SendTestCases(string source, IEnumerable<UnitTestElement> testElem
             foreach (UnitTestElement testElement in testElements)
             {
                 var testCase = testElement.ToTestCase();
+                bool hasFixtureTraits = testCase.Traits.Any(t => t.Name == Constants.FixturesTestTrait);
 
                 // Filter tests based on test case filters
                 if (filterExpression != null && !filterExpression.MatchTestCase(testCase, (p) => TestMethodFilter.PropertyValueProvider(testCase, p)))
                 {
+                    // If test is a fixture test, add it to the list of fixture tests.
+                    if (hasFixtureTraits)
```

Comment:
I think this should be moved out of the filter check but maybe I am missing something in the logic of the flow. For example, if filter is null, you probably still want to show the fixture. Also, we don't really need to process the filtering to add the fixture node.

Reply:
This is not for discovering test for the test explorer. It is filtering out tests which needs to be executed.

Scenarios from test explorer
1. Run all tests
1.1. Filter returns everything (tests + fixtures)
1.2. Tests are reported as before
1.3. All fixtures are moved to a separate list and added at the end

2. Run only some tests
2.1. Filter returns all fixtures + selected tests
2.2. Selected tests are report as before
2.3. All fixtures are moved to a separate list and added at the end

3. Run only fixture test
3.1. Filter returns selected fixture test and is reported (status update for this as user explicitly ran this fixture)
3.2. other fixtures are not added to the list (no status update for these)